lib/Interpreter/Functions/Nextcloud/Get_Tags.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace OCA\FilesScripts\Interpreter\Functions\Nextcloud;
+
+use OC\SystemTag\SystemTagManager;
+use OC\SystemTag\SystemTagObjectMapper;
+use OCA\FilesScripts\Interpreter\RegistrableFunction;
+use OCP\Files\InvalidPathException;
+use OCP\Files\NotFoundException;
+
+/**
+ * `get_tags(Node file): array`
+ *
+ * Gets tags of a file. Returns list of tags.
+ *
+ * ```lua
+ * local tags = get_tags(get_input_files()[1])
+ *
+ * ```
+ */
+class Get_Tags extends RegistrableFunction {
+        use TagsSerializerTrait;
+
+        private SystemTagManager $tagManager;
+        private SystemTagObjectMapper $tagMapper;
+
+        public function __construct(SystemTagManager $tagManager, SystemTagObjectMapper $tagMapper) {
+                $this->tagManager = $tagManager;
+                $this->tagMapper = $tagMapper;
+        }
+
+        public function run($file = []): array {
+                $fileNode = $this->getNode($this->getPath($file));
+
+                if (!$fileNode ) {
+                        return [];
+                }
+
+                $mapping = $this->tagMapper->getTagIdsForObjects($fileNode->getId(), 'files');
+                if (sizeof($mapping) == 0) {
+                        return [];
+                }
+
+                $tags = $this->tagManager->getTagsByIds($mapping[$fileNode->getId()]);
+
+                return $tags;
```

Comment:
Here we need to serialize the tags so the Lua interpreter can convert it into a `Table`. Its also necessary to reindex so that the resulting array is 1-indexed which gives a sequential list in Lua, which makes iterating a bit easier.

```php
$tags = array_values($tags);
return $this->reindex(array_map([$this, 'serializeTag'], $tags));
```